airflow/contrib/hooks/gcp_api_base_hook.py

```diff
@@ -125,11 +125,25 @@ def _get_credentials(self) -> google.auth.credentials.Credentials:
                     google.oauth2.service_account.Credentials.from_service_account_info(
                         keyfile_dict_json, scopes=self.scopes)
                 )
+                project_id = credentials.project_id
             except json.decoder.JSONDecodeError:
                 raise AirflowException('Invalid key JSON.')
 
-        return credentials.with_subject(self.delegate_to) \
-            if self.delegate_to else credentials
+        if self.delegate_to:
+            credentials = credentials.with_subject(self.delegate_to)
+
+        overridden_project_id = self._get_field('project')
```

Comment:
Can we add a test case for this? 

(1) Mock project_id as None from credential object & being overriden by project_id from connection?

Reply:
I added a 8 tests. I am testing a lot of different cases now.

Reply:
👍